fix: the icons background inside stake card

guerrap opened the pull request:
# Summary

Fixes #688

Allow the icon's background-color property to take effect by adjusting the z-index of the parent component.

before:
![image](https://user-images.githubusercontent.com/9011637/158692105-01582b20-44a4-4d98-b762-700ba9a7a636.png)
after:
![after](https://user-images.githubusercontent.com/9011637/158691816-b99816e7-70a8-463a-ab50-736e8f0dd227.png)

  # To Test

1. Open the page `Rewards`
- [ ] Verify that the icons have the white background

berteotti commented on src/components/Pool/LiquidityMiningCampaignView/StakeCard/index.tsx:
```diff
@@ -341,7 +341,7 @@ export default function StakeCard({
   return (
     <>
-      <StyledPositionCard>
+      <StyledPositionCard style={{ zIndex: 1 }}>
```
Can you add this style in the styled component?

berteotti commented on src/components/PositionCard/index.tsx:
```diff
@@ -230,7 +230,7 @@ export default function FullPositionCard({ pair, border }: FullPositionCardProps
 
   return (
-    <StyledPositionCard border={border}>
+    <StyledPositionCard border={border} style={{ zIndex: 1 }}>
```
Can you add this style in the styled component?

MilanVojnovic95 commented:
Tested and looks ok to me.

But we have the same situation on other places like:

Swap
![image](https://user-images.githubusercontent.com/57331244/158905354-50650cf1-77ff-4b85-a107-b591235625c2.png)

Add Liquidity
![image](https://user-images.githubusercontent.com/57331244/158904988-9a4cf0f6-c638-4023-9166-cf4a67eeb93a.png)

Create a liquidity mining pool
![image](https://user-images.githubusercontent.com/57331244/158905114-9ae6a145-52d0-466a-90cd-f346435d334a.png)

guerrap commented:
I have removed the `z-index` from the `StyledPositionCard` inside `PositionCard` and `StakeCard`, and added the `z-index` to the `CurrencyLogo` component; this should fix the background problem at root level. Giving the base component a `z-index: 1` and then in `::after` a `z-index: -1` with the `background` should always place it just beneath the icon. 

![swap](https://user-images.githubusercontent.com/9011637/159080817-07c16c8b-d6e5-41cf-b721-71c37208be68.png)
![liquidity](https://user-images.githubusercontent.com/9011637/159080804-66059c38-ded4-4168-8054-69748467d89b.png)
![pairs](https://user-images.githubusercontent.com/9011637/159080809-6f01e1e7-a14a-44ee-adce-b79fda98059e.png)
![rewards](https://user-images.githubusercontent.com/9011637/159080810-876c78e8-a4a8-4a04-b8f0-cb469c3ab579.png)
![stake](https://user-images.githubusercontent.com/9011637/159080812-b636fa57-90c0-4852-a592-cdbb711023f0.png)

berteotti reviewed (approved):
Nice job!

Mi-Lan reviewed (approved):
This fixes it, GJ!
